kolibri/core/utils/cache.py

```diff
@@ -11,3 +13,85 @@ def __get_process_cache():
 
 
 process_cache = SimpleLazyObject(__get_process_cache)
+_namespace_locks = {}
+
+
+class NamespacedCacheProxy(BaseCache):
+    """
+    Namespaces keys and retains a record of inserted keys for easy clearing of
+    all namespaced keys in the cache
+    """
+
+    def __init__(self, cache, namespace, **params):
+        """
+        :type cache: BaseCache
+        :type namespace: str
+        """
+        params.update(KEY_PREFIX=namespace)
+        super(NamespacedCacheProxy, self).__init__(params)
+        self.cache = cache
+        self._lock = _namespace_locks.setdefault(namespace, RWLock())
+
+    def _get_keys(self):
+        """
+        :rtype: list
+        """
+        return self.cache.get("__KEYS__", default=[])
+
+    def _set_keys(self, keys):
+        """
+        :type keys: list
+        """
+        self.cache.set("__KEYS__", keys)
```

Comment:
Shouldn't the `__KEYS__` be prefixed here by the namespace? Otherwise if we reuse this class elsewhere, the different caches will clear each other's keys?

Reply:
Oh good catch